src/renderAmazonButton.js

```diff
@@ -0,0 +1,70 @@
+/* global amazon_payments_advanced, amazon */
+
+/**
+ * Returns the settings needed to be provided to the amazon.Pay.renderButton().
+ *
+ * @param {string} buttonSettingsFlag Specifies the context of the rendering.
+ * @param {string} checkoutConfig The checkoutConfig with which we will provide Amazon Pay.
+ * @returns {object} The settings to provide the Amazon Pay Button with.
+ */
+const getButtonSettings = ( buttonSettingsFlag, checkoutConfig ) => {
+	const obj = {
+		// set checkout environment
+		merchantId: amazon_payments_advanced.merchant_id,
+		ledgerCurrency: amazon_payments_advanced.ledger_currency,
+		sandbox: amazon_payments_advanced.sandbox === '1' ? true : false,
+		// customize the buyer experience
+		placement: amazon_payments_advanced.placement,
+		buttonColor: amazon_payments_advanced.button_color,
+		checkoutLanguage:
+			amazon_payments_advanced.button_language !== ''
+				? amazon_payments_advanced.button_language.replace( '-', '_' )
+				: undefined,
+		productType:
+			'undefined' !== typeof checkoutConfig.payloadJSON.addressDetails
+				? 'PayAndShip'
+				: 'PayOnly',
+	};
+	return obj;
+};
+
+/**
+ * Renders an Amazon Pay Button on elements identified by btnId.
+ *
+ * @param {string} btnId Selector on where the Amazon Pay button will be rendered on.
+ * @param {string} buttonSettingsFlag Specifies the context of the rendering.
+ * @param {string} checkoutConfig The checkoutConfig with which we will provide Amazon Pay.
+ * @returns {object} The Amazon Pay rendered button.
+ */
+const renderAmazonButton = ( btnId, buttonSettingsFlag, checkoutConfig ) => {
+	let amazonPayBtn = null;
+	const btns = document.querySelectorAll( btnId );
```

Comment:
The param naming can be confusing. The function actually accepts a CSS selector, even though i am not aware of any place we are using it without an id. For the "classic" integration, the loop is not actually necessary, but again it's in place expecting that it will be needed in the near future.

No need to keep it if you'd prefer it removed. We can re establish it when needed.

Reply:
If this will be used in the future to render multiple buttons, with the same `buttonSettingsFlag`, `checkoutConfig` I suppose its Ok. 
Otherwise we should simplify it.